.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,170 @@
+name: Publish NovelRT
+
+on:
+  workflow_dispatch:
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+
+    outputs:
+      tag: ${{ steps.tag_version.outputs.new_tag }}
+      id: ${{ steps.create_release.outputs.id }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: mathieudutour/github-tag-action@v6.0
```

Comment:
If we're using this, we'll have to change our commit message convention to https://www.conventionalcommits.org/en/v1.0.0/ as this is how the tool knows to bump the right version number.

Reply:
Alternatively: we could only run this pipeline when we manually push version tags:
```yml
on:
  create:
    tags:
    - v*
```
That way it will automatically pull the correct version information in the create_release step, and we won't have to figure that out from the commit history. We *will* however have to generate a changelog (which github seems to do automatically: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)

Reply:
Personally, I like the convention based on its readability.

@RubyNova what's your preference here? And you, @FiniteReality ?

Reply:
I'm fine with it too, it's just making sure everything follows it.

Reply:
For example, in PRs we'll likely have to squash or rebase commits in order to ensure that they correctly follow the convention if they don't already.

Reply:
After talking with @RubyNova, it sounds like we're good to switch over. I've enabled squashes for the repo, and will probably disable merge commits once this is in.
Going to take a note to add additional documentation on this as part of #426